.travis.yml

```diff
@@ -2,11 +2,24 @@ language: python
 python:
 - '3.6'
 dist: trusty
-# We're going to ignore all branches except:
-branches:
-  only:
-  - master
-  - production
+
+# Deploy only if:
+#     - it's regular push to the master or production branches
+#     - It's a pull request and:
+#       - PR_AUTOBUILD env. variable is set to "true" or
+#       - the name of the PR branch (head name) is contained in PR_AUTODEPLOY env. variable or
+#       - git commit message contains "[joplin:build]"
+#
+#   Note: PR_AUTOBUILD or PR_AUTODEPLOY variables can be set either in Travis > Settings or
+#   at the env: global: section of this configuration file. See example below.
+#
+
+if: |
+  (branch IN (master, production) AND type = push) OR \
+  (type = pull_request AND env(PR_AUTOBUILD) IN (true, enabled, on)) OR \
+  (type = pull_request AND env(PR_AUTODEPLOY) =~ concat(head_branch) ) OR \
```

Comment:
Do we set PR_AUTOBUILD and PR_AUTODEPLOY to true by default?

Reply:
It seems like this is changing the logic away from what we had before.

It used to be:
* Work in a branch, commit as much as you want, never deploys
* Make a PR, automatically have a PR deployment
* Make changes in the branch you have a PR for, automatically updates the PR environment

Could you explain the reasoning behind adding this complexity?

Reply:
Got it, I just set up the new logic:
(branch IN (master, production) AND type = push) OR (type = pull_request)

Meaning it will build all updates to PRs, master and production branches, leaving regular branches alone.